### ISSUE TITLE
[Bug] Treat `<Link to>` as a non-interactive element

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#313]
I noticed the [a11y-tooltip-interactive-trigger](https://github.com/primer/eslint-plugin-primer-react/blob/main/docs/rules/a11y-tooltip-interactive-trigger.md) hasn't been enabled yet in dotcom - we should enable it because it's super valuable and prevents tooltip misuse!

I [turned it on](https://github.com/github/github/pull/361996) to see what failures would be raised, and noticed 1 false positive scenario where

```
<Tooltip ...>
  <Link to={href} >
</Tooltip>
```

is getting message:

```
Anchor tags without an href attribute are not interactive, therefore they cannot be used as a trigger for a tooltip. Please add an href attribute or use an alternative interactive element instead
```

The rule should be updated to not treat `<Link to>` as a non-interactive element.
